ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -1021,19 +1138,24 @@ mod tests {
         Ok(())
     }
 
-    prop_compose! {
-        fn arb_count_and_chunk()((log_count, log_multiplication_amount) in select(&[(5,5),(7,5),(5,8)])) -> (usize, usize) {
-            (1usize<<log_count, 1usize<<log_multiplication_amount)
-        }
+    fn record_count_strategy() -> impl Strategy<Value = usize> {
+        prop_oneof![1usize..=512, (1usize..=9).prop_map(|i| 1usize << i)]
```

Comment:
I am being stupid and it is 12pm now - is it really testing $$2^{512}$$ records?

Reply:
What it does is:
* 50% of the time it picks a random power of two between 2 and 512 (9 possible values)
* 50% of the time it picks a random integer between 1 and 512 (512 possible values)

Two reasons to focus on powers of two, although I didn't put a huge amount of thought into this (e.g. it just occurred to me to add $2^0$):
* Because we plan to restrict batch sizes to powers of two.
* As the count increases, it makes sense to me to sample the space more sparsely, there's relatively less interesting about testing batch sizes 399, 400, and 401 than about testing batch sizes 7, 8, and 9.

I think it is clearer to have the two `prop_oneof` cases on different lines, but rustfmt insisted on a single line.